src/app/components/popup-host/popup-host.component.ts

```diff
@@ -0,0 +1,35 @@
+import {Component, Input, TemplateRef, ViewChild, ViewContainerRef} from '@angular/core';
+import {CommonModule} from '@angular/common';
+
+@Component({
+    selector: 'app-popup-host',
+    standalone: true,
+    imports: [CommonModule],
+    templateUrl: './popup-host.component.html',
+    styleUrls: ['./popup-host.component.css'],
+})
+export class PopupHostComponent {
+    @ViewChild('content', {read: ViewContainerRef, static: true})
+    private readonly content: ViewContainerRef | undefined;
+
+    @Input() set template(templateRef: TemplateRef<unknown> | null) {
+        if (templateRef) {
+            this.insertTemplate(templateRef);
+        } else {
+            this.closePopup();
+        }
```

Comment:
Чуть чуть попродаю best practics)

Не рекомендую к использованию конструкцию `if/else`, особенно в комбинации нескольких `else` (`if / else if / else`). Вместо этого рекомендую использовать ранний `return`, он более гибки позволяет делать промежуточны вычисления, и, соответсвенно, писать более чистый код)
```suggestion
        if (templateRef) {
            this.insertPopupTemplate(templateRef);
            
            return;
        }
        
        this.onClose();
```

Reply:
Спасибо! поправил